velox/functions/prestosql/aggregates/ChecksumAggregate.cpp

```diff
@@ -0,0 +1,225 @@
+/// a vector.
+/// Checksum will return an order-insensitive checksum of the input vector.
+///
+/// checksum(T)-> varbinary
+class ChecksumAggregate : public exec::Aggregate {
+ public:
+  static const long kPrime64 = 0x9E3779B185EBCA87L;
+
+  explicit ChecksumAggregate() : Aggregate(VARBINARY()) {}
+
+  void finalize(char** /* groups */, int32_t /* numGroups */) override {}
+
+  int32_t accumulatorFixedWidthSize() const override {
+    return sizeof(int64_t);
+  }
+
+  void initializeNewGroups(
+      char** groups,
+      folly::Range<const vector_size_t*> indices) override {
+    for (auto i : indices) {
+      *exec::Aggregate::value<int64_t>(groups[i]) = 0;
+    }
+  }
+
+  void extractValues(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    auto* vector = (*result)->as<FlatVector<StringView>>();
+    VELOX_CHECK(vector);
+    vector->resize(numGroups);
+
+    auto* rawValues = vector->mutableRawValues();
+    for (vector_size_t i = 0; i < numGroups; ++i) {
+      rawValues[i] = StringView(value<char>(groups[i]), sizeof(int64_t));
+    }
+  }
+
+  void extractAccumulators(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    VELOX_CHECK_EQ((*result)->encoding(), VectorEncoding::Simple::FLAT);
+    auto vector = (*result)->as<FlatVector<StringView>>();
+    VELOX_CHECK(
+        vector,
+        "Unexpected type of the result vector: {}",
+        (*result)->type()->toString());
+    vector->resize(numGroups);
+    uint64_t* rawNulls = getRawNulls(vector);
+
+    for (int32_t i = 0; i < numGroups; ++i) {
+      char* group = groups[i];
+      if (isNull(group)) {
+        vector->setNull(i, true);
+      } else {
+        clearNull(rawNulls, i);
+        vector->set(i, StringView(value<char>(group), sizeof(int64_t)));
+      }
+    }
+  }
+
+  void addRawInput(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /**/) override {
+    const auto& arg = args[0];
+    auto hasher = getPrestoHasher(arg->type());
+    BufferPtr hashes =
+        AlignedBuffer::allocate<int64_t>(rows.end(), arg->pool());
+    hasher->hash(arg, rows, hashes);
+    auto rawHashes = hashes->as<int64_t>();
+
+    rows.template applyToSelected([&](vector_size_t row) {
+      if (arg->isNullAt(row)) {
+        computeHashForNull(groups[row]);
+      } else {
+        computeHash(groups[row], rawHashes, row);
+      }
+    });
+  }
+
+  void addIntermediateResults(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /**/) override {
+    VELOX_CHECK_EQ(args[0]->encoding(), VectorEncoding::Simple::FLAT);
+
+    auto vector = args[0]->asUnchecked<FlatVector<StringView>>();
+    auto rawValues = vector->rawValues();
+
+    rows.applyToSelected([&](vector_size_t i) {
+      *value<int64_t>(groups[i]) = *(getHashValue(rawValues[i]));
```

Comment:
I'm wondering if this is correct. It looks like the accumulator for each group is overwritten instead of the existing value being combined with the incoming value.